### PR TITLE
Added Policy class with its relevant parameters. Test case class added but not implemented yet.

### PR DESCRIPTION
**Pull request by YZmunchmunch**
* Added Policy class as a separate window
* Policy class takes in 4 arguments, each of which has their own separate class :
  * All of which are under the package labeled as `policy`
  * `policy` package is within Client package
  ```
  Policy(PolicyName name, CustomDate startdate, Premium premium, Frequency frequency)
  ```
* Added `UniquePolicyList` class as a iterable list which stores the `policy` for each `client`

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -97,6 +97,7 @@ private static Client createEditedClient(Client clientToEdit, EditClientDescript
         Phone updatedPhone = editClientDescriptor.getPhone().orElse(clientToEdit.getPhone());
         Email updatedEmail = editClientDescriptor.getEmail().orElse(clientToEdit.getEmail());
         Address updatedAddress = editClientDescriptor.getAddress().orElse(clientToEdit.getAddress());
+        //UniquePolicyList updatedPolicyList = editClientDescriptor.getPolicyList().orElse(clientToEdit.getAddress());
```
Comment (pangrwa) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:100]: Can I check what is this for?

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 2]
```diff
@@ -42,6 +42,7 @@ public AddCommand parse(String args) throws ParseException {
         Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
         Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
         Address address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get());
+        //UniquePolicyList policyList = new UniquePolicyList();
```
Comment (pangrwa) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:45]: Can I check what is this for too?

File: src/main/java/seedu/address/model/client/policy/CustomDate.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+/**
+ * A class that represents a date in the format of dd.mm.yyyy.
+ */
+public class CustomDate {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "date should only contain numbers, in the format of dd.mm.yyyy";
+
+    public static final String VALIDATION_REGEX = "^\\d{2}\\.\\d{2}\\.\\d{4}$";
```
Comment (pangrwa) [inline src/main/java/seedu/address/model/client/policy/CustomDate.java:18]: If we were to follow this particular REGEX, would we be able to catch exceptions if the user passes in an invalid date?

File: src/main/java/seedu/address/model/client/policy/CustomDate.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof CustomDate // instanceof handles nulls
+                && date.equals(((CustomDate) other).date)); // state check
+    }
+
+    @Override
+    public int hashCode() {
+        return date.hashCode();
```
Comment (pangrwa) [inline src/main/java/seedu/address/model/client/policy/CustomDate.java:65]: Do remember to cover this line in your test case!

File: src/main/java/seedu/address/model/client/policy/Frequency.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    public static final String MESSAGE_CONSTRAINTS = "Frequency can only take on any of the 3 values, "
+            + "\"weekly\", \"monthly\", \"yearly\", and it should not be blank";
+
+    /*
+     * The first character of the frequency must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String[] VALIDATION_REGEX = {"monthly", "quarterly", "yearly"};
+
```
Comment (pangrwa) [inline src/main/java/seedu/address/model/client/policy/Frequency.java:23]: Perhaps we could use an Enum here?

File: src/main/java/seedu/address/model/client/policy/UniquePolicyList.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+     * Removes the equivalent person from the list.
+     * The person must exist in the list.
+     */
+    public void remove(Policy toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new PolicyNotFoundException();
+        }
+    }
+
```
Comment (pangrwa) [inline src/main/java/seedu/address/model/client/policy/UniquePolicyList.java:80]: Can I check what are these commented code for?

File: src/test/java/seedu/address/commons/util/AppUtilTest.java [hunk 7]
```diff
@@ -1,17 +1,11 @@
 package seedu.address.commons.util;
 
-import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static seedu.address.testutil.Assert.assertThrows;
 
 import org.junit.jupiter.api.Test;
 
 public class AppUtilTest {
 
-    @Test
```
Comment (pangrwa) [inline src/test/java/seedu/address/commons/util/AppUtilTest.java:10]: Can I check what is the reason for deleting this block of code?

File: src/main/java/seedu/address/model/client/policy/CustomDate.java (hunk 4 above)
Comment (anunayajoshi) [inline src/main/java/seedu/address/model/client/policy/CustomDate.java:65]: You can fix this codecov by comparing two same dates hashcode and check if its the same.

File: src/test/java/seedu/address/testutil/TypicalPolicies.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            .withFrequency("yearly").build();
+
+
+    // Manually added
+    public static final Policy TRAVEL = new PolicyBuilder().withPolicyName("Travel")
+            .withStartDate("04.05.2023")
+            .withPremium("20.00")
+            .withFrequency("monthly").build();
+
+    // Manually added - Person's details found in {@code CommandTestUtil}
```
Comment (anunayajoshi) [inline src/test/java/seedu/address/testutil/TypicalPolicies.java:33]: Kindly change person to client for consistency with the refactoring

**Review by anunayajoshi — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java (hunk 2 above)
Comment (YZmunchmunch) [reply src/main/java/seedu/address/logic/parser/AddCommandParser.java:45]: This is for when Anun implements `Policy` class into `Client` class!

File: src/main/java/seedu/address/model/client/policy/CustomDate.java (hunk 3 above)
Comment (YZmunchmunch) [reply src/main/java/seedu/address/model/client/policy/CustomDate.java:18]: I believe so. The `CustomDate` class will handle correct format, but wrong date errors.

File: src/test/java/seedu/address/commons/util/AppUtilTest.java (hunk 7 above)
Comment (YZmunchmunch) [reply src/test/java/seedu/address/commons/util/AppUtilTest.java:10]: There was a specific error that only my computer had, so i removed it to be able to commit my branch!

**Review by tituswe — CHANGES_REQUESTED** [review]
Great work, esp the coverage with test cases. Just check for the remaining few then good 2 merge!

**Review by tituswe — APPROVED** [review]

File: src/main/java/seedu/address/model/client/policy/UniquePolicyList.java [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+/**
+ * Represents a list of policies that enforces no two policies in the list have the same identity.
+ * A policy is considered the same as another policy in the list if they have the same identity.
+ * Identity is defined by {@code Policy#isSamePolicy(Policy)}.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Policy#isSamePolicy(Policy)
+ */
+public class UniquePolicyList implements Iterable<Policy> {
```
Comment (pangrwa) [inline src/main/java/seedu/address/model/client/policy/UniquePolicyList.java:24]: How do I determine if the `UniquePolicyList` is empty? `Iterable` does not have an `isEmpty()` method.

**Pull request merged**